ftests/nuxeo-platform-explorer-ftests/src/test/java/org/nuxeo/functionaltests/explorer/ITExplorerAdminTest.java

```diff
@@ -62,4 +136,171 @@ public void testDistribAdminPage() {
+        checkExtensionPoints(true);
+        open(ExplorerHomePage.URL + distribId + "/" + ApiBrowserConstants.LIST_SERVICES);
+        checkServices(true);
+        open(ExplorerHomePage.URL + distribId + "/" + ApiBrowserConstants.LIST_CONTRIBUTIONS);
+        checkContributions(true);
+        open(ExplorerHomePage.URL + distribId + "/" + ApiBrowserConstants.LIST_OPERATIONS);
+        checkOperations(true);
+    }
+
+    // will be easier to maintain as a reference when a specific bundle group is used for explorer...
+    protected void checkPartialBundleGroup(String distribName) {
+        Locator.findElementWaitUntilEnabledAndClick(By.linkText(distribName));
+        BundleGroupArtifactPage apage = asPage(BundleGroupArtifactPage.class);
+        apage.checkDocumentationText(null);
+        apage.checkSubGroup(null);
+        apage.checkBundle("org.nuxeo.apidoc.core");
+        apage.checkBundle("org.nuxeo.apidoc.repo");
+        apage.checkBundle("org.nuxeo.apidoc.webengine");
+    }
+
+    @Test
+    public void testLivePartialDistribExportAndImport() {
+        String distribName = "my-partial-server";
+        open(DistribAdminPage.URL);
+        String version = asPage(DistribAdminPage.class).saveCurrentLiveDistrib(distribName, true);
+        String distribId = getDistribId(distribName, version);
+        asPage(ExplorerHomePage.class).checkPersistedDistrib(distribId);
+        checkPartialDistrib(distribName, distribId);
+
+        // check importing it back
+        open(DistribAdminPage.URL);
+        String filename = getDistribExportName(distribId);
+        File file = asPage(DistribAdminPage.class).exportFirstPersistedDistrib(downloadDir, filename);
+
+        open(DistribAdminPage.URL);
+        String newDistribName = "partial-imported-server";
+        String newVersion = "1.0.0";
+        asPage(DistribAdminPage.class).importPersistedDistrib(file, newDistribName, newVersion, null);
+        open(ExplorerHomePage.URL);
+        String newDistribId = getDistribId(newDistribName, newVersion);
+        asPage(ExplorerHomePage.class).checkPersistedDistrib(newDistribId);
+        checkPartialDistrib(distribName, newDistribId);
+    }
+
+    protected void createSampleZip(String sourceDirPath, String zipFilePath, boolean addMarker) throws IOException {
+        Path p = Files.createFile(Paths.get(zipFilePath));
+        try (ZipOutputStream zs = new ZipOutputStream(Files.newOutputStream(p))) {
+            if (addMarker) {
+                ZipEntry zipEntry = new ZipEntry(".nuxeo-archive");
+                zs.putNextEntry(zipEntry);
+                zs.closeEntry();
+            }
+            // read paths from reference file as NuxeoArchiveReader requires a given order and extra info
+            List<String> lines = Files.readAllLines(Paths.get(sourceDirPath, "entries.txt"));
+            for (int i = 0; i < lines.size(); i++) {
+                String path = lines.get(i);
```

Comment:
I'd rather not change these logics now because i'm getting the next next line sometimes (to fill extra info on the zip entry), and because i fought with NuxeoArchiveReader expected format.
But i'll keep that in mind :+1:

Reply:
Well my suggestion doesn't change the logic, just how we write the code. It does the same thing.

Reply:
Yes i got that, still in some cases i have an extra ++ so i kind of need the index here not to complicate things (at least for me)

Reply:
After all see related commit with #4059